app/assets/javascripts/tasks.js.coffee

```diff
@@ -58,26 +58,16 @@ $ ->
     $("table.tasks").bootstrapTable('hideColumn', 'id')
 
 
-    $(document).on "click", "#complete", ->
-      scope = $(this).prop('value')
+    $(document).on "change", "#complete", ->
```

Comment:
This could be a single handler. 
```
$(document).on 'change', '#complete, #all_tasks', ->
  status = if $(this).prop('checked') then 'all' else 'mine'
  status = if $('#complete').prop('checked') then 'complete' else 'incomplete'
```